pkg/controller/v1alpha2/experiment/experiment_controller.go

```diff
@@ -80,17 +108,251 @@ type ReconcileExperiment struct {
+}
+
+func (r *ReconcileExperiment) ReconcileExperiment(instance *experimentsv1alpha2.Experiment) error {
+
+	var err error
+	logger := log.WithValues("Experiment", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
+	trials := &trialsv1alpha2.TrialList{}
+	labels := map[string]string{"experiment": instance.Name}
+	lo := &client.ListOptions{}
+	lo.MatchingLabels(labels).InNamespace(instance.Namespace)
+
+	err = r.List(context.TODO(), lo, trials)
+	if err != nil {
+		logger.Error(err, "Trial List error")
+		return err
+	}
+	if len(trials.Items) > 0 {
+		err := util.UpdateExperimentStatus(instance, trials)
+		if err != nil {
+			logger.Error(err, "Update experiment status error")
+			return err
+		}
+	}
+	reconcileRequired := !instance.IsCompleted()
+	if reconcileRequired {
+		r.ReconcileTrials(instance)
+	}
+	return err
+}
+
+func (r *ReconcileExperiment) ReconcileTrials(instance *experimentsv1alpha2.Experiment) error {
+
+	var err error
+	logger := log.WithValues("Experiment", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
+	parallelCount := 0
+
+	if instance.Spec.ParallelTrialCount != nil {
+		parallelCount = *instance.Spec.ParallelTrialCount
+	} else {
+		parallelCount = 3
+	}
+	activeCount := instance.Status.TrialsPending + instance.Status.TrialsRunning
+	completedCount := instance.Status.TrialsSucceeded + instance.Status.TrialsFailed + instance.Status.TrialsKilled
+
+	if activeCount > parallelCount {
+		deleteCount := activeCount - parallelCount
+		if deleteCount > 0 {
+			//delete 'deleteCount' number of trails. Sort them?
+			logger.Info("DeleteTrials", "deleteCount", deleteCount)
+			err = r.deleteTrials(instance, deleteCount)
+		}
+
+	} else if activeCount < parallelCount {
+		requiredActiveCount := 0
+		if instance.Spec.MaxTrialCount == nil {
+			requiredActiveCount = parallelCount
+		} else {
+			requiredActiveCount = *instance.Spec.MaxTrialCount - completedCount
+			if requiredActiveCount > parallelCount {
+				requiredActiveCount = parallelCount
+			}
+		}
+
+		addCount := requiredActiveCount - activeCount
+		if addCount < 0 {
+			logger.Info("Invalid setting", "requiredActiveCount", requiredActiveCount, "MaxTrialCount",
+				*instance.Spec.MaxTrialCount, "CompletedCount", completedCount)
+			addCount = 0
+		}
+
+		//create "addCount" number of trials
+		logger.Info("CreateTrials", "addCount", addCount)
+		err = r.createTrials(instance, addCount)
+
+	}
+
+	return err
+
+}
+
+type TrialTemplateParams struct {
+	Experiment      string
+	Trial           string
+	NameSpace       string
+	HyperParameters []*apiv1alpha2.ParameterAssignment
+}
+
+func (r *ReconcileExperiment) createTrials(instance *experimentsv1alpha2.Experiment, addCount int) error {
+
+	logger := log.WithValues("Experiment", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
+	trials, err := util.GetSuggestions(instance, addCount)
```

Comment:
@johnugeorge 
It can run more than one Suggestion at once, since we name it `GetSuggestions`?

Reply:
Can you please explain more?

Reply:
I mean you have named this function `GetSuggestions`. Is it make sense? I thought we should name it `GetSuggestion`.